scalafix-core/shared/src/main/scala/scalafix/internal/rule/RuleCtxImpl.scala

```diff
@@ -87,6 +88,20 @@ case class RuleCtxImpl(tree: Tree, config: ScalafixConfig) extends RuleCtx {
   def addLeft(tok: Token, toAdd: String): Patch = Add(tok, toAdd, "")
   def addLeft(tree: Tree, toAdd: String): Patch =
     toks(tree).headOption.fold(Patch.empty)(addLeft(_, toAdd))
+  def trimLeft(token: Token, includeNewLine: Boolean = true): Patch =
+    doRemoveTokens(
+      tokenList.leading(token).takeWhile(canBeTrimmed(_, includeNewLine)))
+  def trimRight(token: Token, includeNewLine: Boolean = true): Patch =
+    doRemoveTokens(
+      tokenList.trailing(token).takeWhile(canBeTrimmed(_, includeNewLine)))
+  def trim(token: Token, includeNewLine: Boolean = true): Patch =
+    trimLeft(token, includeNewLine) + trimRight(token, includeNewLine)
+  private def canBeTrimmed(tk: Token, includeNewLine: Boolean): Boolean =
+    tk match {
+      case Space() | Tab() | FF() => true
+      case Newline() if includeNewLine => true
```

Comment:
I'm wondering if we should limit this to a single newline. A typical use case if you want to remove a statement then you usually want to keep the leading / trailing blank lines. What do you think?

Reply:
Just to understand, given this example:

```scala
val x = 42

  "foo"

val y = "bar"
```

would the current implementation be able to produce this diff?

```diff
val x = 42
-
-  "foo"

val y = "bar"
```

so that the resulting code is

```scala
val x = 42

val y = "bar"
```

Reply:
I think the implementation should produce this diff
```diff
val x = 42

-  "foo"

val y = "bar"
```

That would leave two blank lines, but it would also preserve a blank line in the following case instead of remove it
```diff
val x = 42

-  "foo"
val y = "bar"
```

Reply:
Right. Probably the job of taking care of multiple blank lines belongs to a formatter

Reply:
Yes, scalafmt would at least combine the two blank lines into a single blank line.